napari/layers/surface/_tests/test_surface.py

```diff
@@ -35,7 +34,7 @@ def test_random_3D_surface():
     assert layer._data_view.shape[1] == 2
     assert layer._view_vertex_values.ndim == 1
 
-    layer.dims.ndisplay = 3
+    layer._slice_dims(ndisplay=3)
```

Comment:
This kinda makes me question the choice of making dims private for now. It's very clunky. What's the long term vision here? Don't we want individual layers to be able to have different ndisplays?

Reply:
long term vision is that layers shouldn't know about ndisplay as it relates to slicing, and we'll probably have some public slice method that is very clean, but ndisplay never gets stored on the layer itself and the sliced data never gets stored on the layer itself